blueocean-events/src/main/java/io/jenkins/blueocean/events/BlueMessageEnricher.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.blueocean.events;
+
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.ItemGroup;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.service.embedded.rest.OrganizationImpl;
+import io.jenkins.blueocean.service.embedded.rest.PipelineImpl;
+import jenkins.model.ParameterizedJobMixIn;
+import org.jenkins.pubsub.EventProps;
+import org.jenkins.pubsub.Events;
+import org.jenkins.pubsub.JobChannelMessage;
+import org.jenkins.pubsub.Message;
+import org.jenkins.pubsub.MessageEnricher;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+
+import javax.annotation.Nonnull;
+
+/**
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+@Extension
+public class BlueMessageEnricher extends MessageEnricher {
+    
+    enum BlueEventProps {
+        blueocean_job_rest_url,
+        blueocean_job_pipeline_name,
+        blueocean_job_branch_name,
+    }
+    
+    @Override
+    public void enrich(@Nonnull Message message) {
```

Comment:
@tfennelly There is nothing today that will give you Link for a given job, but something we can do what you are asking thru proper abstraction that works with any rest implementation. What I was saying is that, if you need it now, you can go ahead with the code sample I gave above to get instance of BluePipeline and get Link and Organization name from it. Maybe I should do it now, and you can remove dependency on rest-impl and do it right way?